### PR TITLE
do not add the API suffix macro to function definitions

### LINKED ISSUE #123: Function definitions shouldn't use __attribute__((deprecated))
The function definitions in the code generated for the ICD loader dispatch use the prefix/suffix preprocessor macros for marking APIs as deprecated. These macros potentially expand to include GCC's `__attribute__((deprecated))`, which isn't valid on function definitions. As noted in #122, this isn't an issue in practice for the ICD loader since we currently always support deprecated functions and therefore define the `CL_USE_DEPRECATED_OPENCL_*_APIS` macros that disable this.

### PR DESCRIPTION
**Pull request by bashbaug**
Fixes #123 

This PR removes the API suffix macro from all ICD dispatch functions, including generated functions and manually written functions.  The API suffix macro should be included on function headers but not on function definitions.

### CODE REVIEW

**Review by alycm — APPROVED** [review]
LGTM, the rationale in https://github.com/KhronosGroup/OpenCL-ICD-Loader/issues/123 makes sense.

**Comment by Kerilk** [comment]
We would be saving one merge if we merged/refactored this on top of the layers, as this already conflicts with `master`, and is sure to conflict with the layers.

**Review by Kerilk — APPROVED** [review]
LGTM

**Comment by vladimirlaz** [comment]
@bashbaug  the macro has been removed from OpenCL headers (https://github.com/KhronosGroup/OpenCL-Headers/commit/9d46d9580cc469757ceb8d89842366f6c9a20093) and the build with the latest OpenCL headers failing.

Could you submit the PR?

**Comment by bashbaug** [comment]
I merged https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/142 to unblock the builds, but we really should get this change merged so we don't run into similar issues again.

**Review by kpet — APPROVED** [review]
LGTM

**Comment by bashbaug** [comment]
This has three approvals and has been open for a very long time, so merging.

**Pull request merged**
